huma.go

```diff
@@ -696,12 +696,24 @@ func Register[I, O any](api API, op Operation, handler func(context.Context, *I)
 				return
 			}
 
-			pv, err := parseInto(ctx, f, value, nil, *p)
+			var receiver = f
+			if f.Addr().Type().Implements(reflect.TypeFor[ParamWrapper]()) {
+				receiver = f.Addr().Interface().(ParamWrapper).Receiver()
+				// Feels like a hack, but we need to override the type of the wrapper from the cache
+				// There might a be a better way
+				p.Type = receiver.Type()
```

Comment:
I suspect we want to find the right type in `findParams` at https://github.com/danielgtaylor/huma/blob/main/huma.go#L102 by calling `Receiver()` once on a dummy value, then we don't have to do this check/swap on each request.